platforms/core-configuration/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/TaskContainerExtensions.kt

```diff
@@ -164,15 +165,23 @@ private constructor(
      * @see [TaskProvider.configure]
      */
     operator fun String.invoke(configuration: Task.() -> Unit): TaskProvider<Task> =
-        this().apply { configure(configuration) }
+        named(this).apply { configure(configuration) }
 
     /**
      * Locates a task by name, without triggering its creation or configuration, failing if there is no such task.
      *
      * @see [TaskContainer.named]
      */
-    operator fun String.invoke(): TaskProvider<Task> =
-        container.named(this)
+    @Deprecated("Use named(String) instead.", ReplaceWith("named(\"...\")"))
```

Comment:
❌ We can fully automate replacing this properly with different syntax.
```suggestion
    @Deprecated("Use `named(String)` instead.", ReplaceWith("named(this)"))
```
Though I would verify that this actually works, I believe this is correct according to the [documentation](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-replace-with/expression.html). 

💭 I also changed the message to use backticks for clarity.

Reply:
Replacing this with `named(this)` is a good idea. It doesn't work automatically in IntelliJ, but you can get there.
```
tasks { 
   "help"()
}
```
This will say it's deprecated, but provide no quick fix.  

If you move to the end of the line, it suggests replacing the operator with the explicit method call, so `() -> invoke()`.  Selecting this does nothing.

If you manually rewrite it to `"help".invoke()`, it's still deprecated.  The quick fix works this time and switches it to `named("help")`.

I'm not sure about the backticks for the same reason as above in the deprecation message. If IntelliJ rendered them as code, that would be OK, but it ends up just being noise.